src/main/java/mate/academy/bookshop/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package mate.academy.bookshop.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.category.CategoryDto;
+import mate.academy.bookshop.dto.category.CategoryRequestDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.exception.NotUniqueValueException;
+import mate.academy.bookshop.mapper.CategoryMapper;
+import mate.academy.bookshop.model.Category;
+import mate.academy.bookshop.repository.CategoryRepository;
+import mate.academy.bookshop.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll().stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find category by id: " + id));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CategoryRequestDto categoryDto) {
+        if (categoryRepository.existsByName(categoryDto.getName())) {
+            throw new NotUniqueValueException("Category name must by unique");
+        }
+        Category category = categoryRepository.save(categoryMapper.toEntity(categoryDto));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryRequestDto categoryDto) {
+        getById(id);

```

Comment:
Value that you return is ignored, better to create separate method to check is id valid